sdk/python/lib/pulumi/_types.py

```diff
@@ -77,44 +98,45 @@
 #           ...
 #
 # This latter (more verbose) declaration is equivalent to the former (simpler) declaration;
-# the `@pulumi.input_type` processes the class and essentially transforms the former declaration
-# into the latter declaration.
+# the `@pulumi.input_type` processes the class and transforms the former declaration into the
+# latter declaration.
 #
-# The former (simpler) declaration is nice syntactic sugar to use when declaring these by hand,
+# The former (simpler) declaration is syntactic sugar to use when declaring these by hand,
 # e.g. when writing a dynamic provider that has nested inputs/outputs. The latter declaration isn't
 # as pleasant to write by hand and is closer to what we emit in our provider codegen. The benefit
 # of the latter (more verbose) form is that it allows docstrings to be specified on the Python
 # property getters, which will show up in IDE tooltips when hovering over the property.
 #
-# Note the property getter/setter functions are empty in the latter (more verbose) declaration.
+# Note the property getter/setter functions are empty in the more verbose declaration.
```

Comment:
I'd like to understand the reason behind leaving this function empty and having it be automatically replaced. Since we are generating the code anyway, why not generate the actual implementation? Are we doing this just to avoid the verbosity?

Reply:
Yeah, I was wondering if this was "too much magic": https://github.com/pulumi/pulumi-aws/commit/a77712011c3dc7577a2737b5b3004ee529ea5dc6#r41448321

The motivation for allowing empty bodies was so you wouldn't have to specify names twice (if writing this more verbose form by hand): once in the declaration of the function and again inside the function's body. But I'm definitely open to just emitting the actual implementation in the generated code, where that doesn't matter (and too much "magic" is confusing / hard to debug).

If we did leave the bodies empty, we could theoretically generate faster lookup code in some cases. That's not being done currently, but in the cases where we need to translate the property names a couple times per invocation, we could generate that code ahead-of-time, rather than doing the translations at runtime. But I doubt this would make much of a measurable difference in practice.

Reply:
Yeah, it's nice that this option exists for the cases where a provider is being hand-written but for the generated code I vote for emitting the actual implementation for the sake of simplicity and clarity.

Reply:
That sounds good to me. 👍